### ISSUE TITLE
(1146, "Table 'sale.predict_product' doesn't exist")

### ISSUE THREAD

**Issue by RahulKumarSavariya** [rshrc/sales-forecast-ml#4]
I click on the predict button in the product form. After that i receive this error. Any solution for that.. 
django == 2.2.5

▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/mysql/base.py in execute
            return self.cursor.execute(query, args) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/MySQLdb/cursors.py in execute
        res = self._query(query) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/MySQLdb/cursors.py in _query
        db.query(q) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/MySQLdb/connections.py in query
        _mysql.connection.query(self, query) …
▶ Local vars
The above exception ((1146, "Table 'sale.predict_product' doesn't exist")) was the direct cause of the following exception:
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/core/handlers/exception.py in inner
            response = get_response(request) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/core/handlers/base.py in _get_response
                response = self.process_exception_by_middleware(e, request) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/core/handlers/base.py in _get_response
                response = wrapped_callback(request, *callback_args, **callback_kwargs) …
▶ Local vars
/home/emxcel/Projects_python/sales-forecast-ml-master/predict/views.py in product_describe_view
            product_detail = product_form.save() …
▶ Local vars
/home/emxcel/Projects_python/sales-forecast-ml-master/predict/forms.py in save
            product.save() …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/base.py in save
                       force_update=force_update, update_fields=update_fields) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/base.py in save_base
                force_update, using, update_fields, …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/base.py in _save_table
            result = self._do_insert(cls._base_manager, using, fields, update_pk, raw) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/base.py in _do_insert
                               using=using, raw=raw) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/manager.py in manager_method
                return getattr(self.get_queryset(), name)(*args, **kwargs) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/query.py in _insert
        return query.get_compiler(using=using).execute_sql(return_id) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/models/sql/compiler.py in execute_sql
                cursor.execute(sql, params) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/utils.py in execute
            return super().execute(sql, params) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/utils.py in execute
        return self._execute_with_wrappers(sql, params, many=False, executor=self._execute) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/utils.py in _execute_with_wrappers
        return executor(sql, params, many, context) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/utils.py in _execute
                return self.cursor.execute(sql, params) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/utils.py in __exit__
                raise dj_exc_value.with_traceback(traceback) from exc_value …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/utils.py in _execute
                return self.cursor.execute(sql, params) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/django/db/backends/mysql/base.py in execute
            return self.cursor.execute(query, args) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/MySQLdb/cursors.py in execute
        res = self._query(query) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/MySQLdb/cursors.py in _query
        db.query(q) …
▶ Local vars
/home/emxcel/anaconda3/envs/oenv/lib/python3.6/site-packages/MySQLdb/connections.py in query
        _mysql.connection.query(self, query) …
▶ Local vars

**Comment by rshrc** [comment]
Hello @RahulKumarSavariya it seems to be a migration error. Did you run the migrations commands?
*The above exception ((1146, "Table 'sale.predict_product' doesn't exist"))* this error means that the table was not created in the first place.

**Comment by RahulKumarSavariya** [comment]
yes, i run the command migration command.  output :  " No change detect. " In your directory i did not find the migration folder. so I run migration command with app run. It make the migration folder and file. I connected  this project with sql database. After click the predict button,  it's gave the error " predict_product can't be null". I  make some change in the preduct_product column (null =True, blank =True).  Thanks.

**Comment by RahulKumarSavariya** [comment]
Thanks @rshrc  .  Actually i didn't write the app name in migration command.  After, click the predict button, i received the error "(1048, "Column 'predict_product' cannot be null") ".  I modify the predict_product column  null and blank to true. Can you explain it why ? 
![My_sql_predict](https://user-images.githubusercontent.com/35863129/70121934-a016ac00-1695-11ea-9555-044b9f27cee9.png)

**Comment by rshrc** [comment]
Yes, the migration problem is with the django version. It does not migrate the apps with **python manage.py makemigrations**

You need to run **python manage.py makemigrations predict**

This should create the predict.product table
